src/main/java/domain/ExpressionUtils.java

```diff
@@ -0,0 +1,25 @@
+package domain;
+
+import java.util.regex.Pattern;
+
+public class ExpressionUtils {
+    public static final String SINGLE_NUMBER_REGEX = "[0-9]";
+    public static final String DEFAULT_REGEX = "^(((\\d)(\\,|:))+\\d$)";
+    public static final String CUSTOM_REGEX = "//(.)\\n(.*)";
+
+    public static boolean isEmptyExpression(String expressionStr) {
+        return expressionStr == null || expressionStr.isEmpty();
+    }
+
+    public static boolean isSingleNumberExpression(String expressionStr) {
+        return Pattern.matches(SINGLE_NUMBER_REGEX, expressionStr);
+    }
+
+    public static boolean isDefaultExpression(String expressionStr) {
+        return Pattern.matches(DEFAULT_REGEX , expressionStr);
+    }
+
+    public static boolean isCustomExpression(String expressionStr) {
+        return Pattern.compile(CUSTOM_REGEX).matcher(expressionStr).find();
```

Comment:
현재 ExpressionUtils의 메서드들을 호출할때마다 Pattern.matches() or Pattern.compile()을 호출하고 있네요 :)
Pattern.matches 내부 구현은 다음과 같은데요 :)
```
public static boolean matches(String regex, CharSequence input) {
    Pattern p = Pattern.compile(regex);
    Matcher m = p.matcher(input);
    return m.matches();
}
```

Pattern.compile 내부 구현을 보면
```
// Pattern
public static Pattern compile(String regex) {
    return new Pattern(regex, 0);
}
```

매번 메서드 호출시 성능이 비싼 Pattern 인스턴스를 생성하고 있기에
Pattern 인스턴스는 한번 쓰고 버려지기 때문에 성능적으로 좋지 않다고 생각해요 :)
자바 doc에서도 한번 컴파일 후 재사용 할것을 권장하고 있는 것 같아요!

```
If a pattern is to be used multiple times, compiling it once and reusing
it will be more efficient than invoking this method each time
```

불필요한 객체 생성을 피하도록 개선해주시면 좋을 것 같아요 😄

Reply:
앞으론 이런 기능들을 사용할때 내부구조까지 고려해보도록하겠습니다!
몰랐던 기능을 알려주셔서 감사합니다 ㅎㅎ